src/evaluators/SwitchStatement.js

```diff
@@ -131,16 +132,17 @@ function AbstractCaseBlockEvaluation(
       invariant(selectionResult instanceof AbstractValue);
```

Comment:
I think the real problem here is that selection result has not been refined with the current path conditions. Instead of "!selectionResult.mightNotBeTrue()" write "Path.implies(selectionResult)" (likewise for mightNotBeFalse).

Reply:
You were right! Thanks for the help.